utils/analyticsmetrics.go

```diff
@@ -171,3 +171,20 @@ func (ams *AnalyticsMetricsService) CreateXscAnalyticsGeneralEventFinalizeFromAu
 		XscAnalyticsBasicGeneralEvent: basicEvent,
 	}
 }
+
+func (ams *AnalyticsMetricsService) UpdateXscAnalyticsGeneralEventFinalizeWithTotalScanDuration() {
+	totalDuration := time.Since(ams.GetStartTime())
+	ams.finalizeEvent.TotalScanDuration = totalDuration.String()
+}
+
+func (ams *AnalyticsMetricsService) UpdateXscAnalyticsGeneralEventFinalizeStatus(status xscservices.EventStatus) {
+	ams.finalizeEvent.EventStatus = status
+}
+
+func (ams *AnalyticsMetricsService) AddScanFindingsToXscAnalyticsGeneralEventFinalize(findingsAmount int) {
+	ams.finalizeEvent.TotalFindings += findingsAmount
+}
+
+func (ams *AnalyticsMetricsService) SetShouldReportEvents(shouldReportEvents bool) {
+	ams.shouldReportEvents = shouldReportEvents
+}
```

Comment:
in Frogbot the analytics report takes several steps, and in some of them some calculations need to be performed.
If we calculated that we should report but the report failed - we dont want to perform the next steps and the final report.
We can check for the existence of the MSI each time but I think it looks more understandable when we have it in this field (id we should report and the initial report fails- the value remains true and it is confusing. I'd rather set this to false in case we fail the initial report)